### PR TITLE
Tango ROS nodelet to node

### PR DESCRIPTION
**Pull request by mcopejans**

### CODE REVIEW

File: TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java [hunk 1]
```diff
@@ -84,80 +91,117 @@ public void onNewMessage(Int8 status) {
...
+                mCallbackListener.onSaveMapServiceCallFinish(false, e.getMessage());
+            }
+        });
+
+        return true;
+    }
+
+    public Boolean getMapUuids() throws ServiceNotFoundException {
+        ServiceClient<GetMapUuidsRequest, GetMapUuidsResponse> saveMapService =
+                mConnectedNode.newServiceClient(SAVE_MAP_SRV_NAME, SaveMap._TYPE);
```
Comment (PerrineAguiar) [inline TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java]: nit: GET_MAP_UUIDS_SRV_NAME

File: TangoRosStreamer/nodelet_manager/src/main/java/eu/intermodalics/nodelet_manager/TangoInitializationHelper.java [hunk 2]
```diff
@@ -122,70 +122,70 @@ public static final int loadTangoSharedLibrary() {
         if (!(new File(basePath).exists())) {
             basePath = "/data/data/com.projecttango.tango/libfiles/";
         }
-        Log.i("TangoInitializationHelper", "basePath: " + basePath);
+        Log.i("TangoInitializationHelp", "basePath: " + basePath);
```
Comment (PerrineAguiar) [inline TangoRosStreamer/nodelet_manager/src/main/java/eu/intermodalics/nodelet_manager/TangoInitializationHelper.java]: Why not keeping TangoInitializationHelper?

File: tango_ros_common/tango_ros_native/include/tango_ros_native/tango_ros_node.h [hunk 3]
```diff
@@ -151,9 +156,14 @@ class TangoRosNode {
   // Function called when one of the dynamic reconfigure parameter is changed.
   // Updates the publisher configuration consequently.
   void DynamicReconfigureCallback(PublisherConfig &config, uint32_t level);
+  bool GetMapName(const tango_ros_messages::GetMapName::Request& req,
```
Comment (PerrineAguiar) [inline tango_ros_common/tango_ros_native/include/tango_ros_native/tango_ros_node.h]: Add doc?

File: tango_ros_common/tango_ros_native/include/tango_ros_native/tango_ros_node.h [hunk 4]
```diff
@@ -151,9 +156,14 @@ class TangoRosNode {
...
+                  tango_ros_messages::GetMapName::Response &res);
+  bool GetMapUuids(const tango_ros_messages::GetMapUuids::Request &req,
```
Comment (PerrineAguiar) [inline tango_ros_common/tango_ros_native/include/tango_ros_native/tango_ros_node.h]: Add doc?

File: TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java [hunk 5]
```diff
@@ -84,80 +91,117 @@ public void onNewMessage(Int8 status) {
         });
     }
 
-    public boolean callSaveMapService(String mapName) {
+    public boolean wrapCallROSService(String serviceName, Callable<Boolean> callable) {
```
Comment (PerrineAguiar) [inline TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java]: Can we make this private?

File: TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java [hunk 6]
```diff
@@ -84,80 +91,117 @@ public void onNewMessage(Int8 status) {
...
         return true;
     }
 
-    public boolean callTangoConnectService(final byte connectRequest) {
-        if (mConnectedNode == null) {
-            // Using system logger since node logger is not ready yet.
-            System.out.println("Client node not ready: " + TANGO_CONNECT_SRV_NAME);
-            return false;
-        }
+    public Boolean tangoConnect(final byte connectRequest) throws ServiceNotFoundException {
```
Comment (PerrineAguiar) [inline TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java]: Can we make this private?

File: TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java [hunk 7]
```diff
@@ -84,80 +91,117 @@ public void onNewMessage(Int8 status) {
...
-        } catch (Exception e) {
-            mLog.error("Error while calling: " + TANGO_CONNECT_SRV_NAME + ", message: " + e.getMessage());
-            return false;
-        }
         return true;
     }
 
-    private String BuildTangoRosNodeNamespaceName(String paramName) {
-        return "/" + TangoRosNode.NODE_NAME + "/" + paramName;
+    public Boolean saveMap(String mapName) throws ServiceNotFoundException {
```
Comment (PerrineAguiar) [inline TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java]: Same here, can we make this private?

File: TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java [hunk 8]
```diff
@@ -84,80 +91,117 @@ public void onNewMessage(Int8 status) {
...
+            @Override
+            public void onFailure(RemoteException e) {
+                mCallbackListener.onSaveMapServiceCallFinish(false, e.getMessage());
+            }
+        });
+
+        return true;
+    }
+
+    public Boolean getMapUuids() throws ServiceNotFoundException {
```
Comment (PerrineAguiar) [inline TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/TangoServiceClientNode.java]: Can we make this private?

File: TangoRosStreamer/nodelet_manager/src/main/java/eu/intermodalics/nodelet_manager/TangoInitializationHelper.java (hunk 2 above)
Comment (mcopejans) [inline TangoRosStreamer/nodelet_manager/src/main/java/eu/intermodalics/nodelet_manager/TangoInitializationHelper.java]: Android studio complained on this. The tag can have maximum 23 characters. Would you prefer a different tag? E.g. just TangoInitialization?

Comment (PerrineAguiar) [inline TangoRosStreamer/nodelet_manager/src/main/java/eu/intermodalics/nodelet_manager/TangoInitializationHelper.java]: Ok, it is fine like this, no need to change. I was just wondering.

**Comment by mcopejans** [comment]
@PerrineAguiar @smits Thanks for your reviews. I addressed all comments. I'm not going to merge this into the base branch since this will get a huge PR otherwise. Feel free to merge whenever it suits you.

**Pull request merged**